### PR TITLE
Show Approximate Fiat amount in the Crypto flow using TGB API

### PR DESCRIPTION
**Pull request by SudharakaP**
Resolve https://github.com/opencollective/opencollective/issues/5448

⚠️ Require: https://github.com/opencollective/opencollective-api/pull/7541

image

### CODE REVIEW

File: components/contribution-flow/StepDetailsCrypto.js [hunk 1]
```diff
@@ -71,30 +51,27 @@ const validateCryptoCurrencyAmount = (touched, amount, minimumAmount, cryptoCurr
...
   const collectiveCurrency = collective.currency;
   const dispatchChange = (field, value) => {
     onChange({ stepDetails: { ...data, [field]: value }, stepSummary: null });
   };
 
-  const storeCryptoExchangeRate = async (cryptoCurrency, collectiveCurrency) => {
-    const exchangeRate = await getCryptoExchangeRate(cryptoCurrency, collectiveCurrency);
-    setCryptoExchangeRate(exchangeRate);
-  };
+  const { data: fxData } = useQuery(cryptoExchangeRateQuery, {
```
Comment (znarf) [inline components/contribution-flow/StepDetailsCrypto.js:64]: `fxData` is not a great name

File: components/contribution-flow/StepDetailsCrypto.js [hunk 2]
```diff
@@ -71,30 +51,27 @@ const validateCryptoCurrencyAmount = (touched, amount, minimumAmount, cryptoCurr
 
 const StepDetailsCrypto = ({ onChange, data, collective }) => {
   const intl = useIntl();
+
   const [selectedCryptoCurrency, setSelectedCryptoCurrency] = useState(data.currency);
   const [amount, setAmount] = useState(data.amount);
   const [convertedAmount, setConvertedAmount] = useState(null);
-  const [cryptoExchangeRate, setCryptoExchangeRate] = useState(null);
   const [touched, setTouched] = useState(false);
   const collectiveCurrency = collective.currency;
```
Comment (znarf) [inline components/contribution-flow/StepDetailsCrypto.js:59]: What the benefit of `collectiveCurrency` compared to `collective.currency`

File: components/contribution-flow/StepDetailsCrypto.js [hunk 3]
```diff
@@ -71,30 +51,27 @@ const validateCryptoCurrencyAmount = (touched, amount, minimumAmount, cryptoCurr
...
-  }, []);
-
-  useEffect(() => {
-    if (cryptoExchangeRate) {
-      setConvertedAmount(amount * cryptoExchangeRate);
+    if (fxData) {
+      setConvertedAmount(amount * fxData.cryptoExchangeRate);
     }
-  }, [amount, cryptoExchangeRate]);
+  }, [amount, fxData]);
```
Comment (znarf) [inline components/contribution-flow/StepDetailsCrypto.js:74]: What kind of failure are you taking care of? What happens if `fxData.cryptoExchangeRate` is null?

File: components/contribution-flow/StepDetailsCrypto.js (hunk 1 above)
Comment (SudharakaP) [reply components/contribution-flow/StepDetailsCrypto.js]: I've changed this to `cryptoExchangeRateData` instead.

**Pull request closed without merging**